### DISCUSSION TITLE
How to do an OR query on multiple fields?

### DISCUSSION THREAD

**Discussion by lwhiteley** [mswjs/data#192] [Q&A] [answered]
Hi, im trying to do a search that can consider multiple fields but not clear on how to do this

for example

```js
const searchPhrases = "John Doe".split(' '); // ['John', 'Doe']
const userMatches = db.users.findMany({
  where: {
    firstName: {
      contains: searchPhrases, // check if firstName contains one of the values
    },
    lastName: {
      contains: searchPhrases, // check if lastName contains one of the values
    },
  },
})
```

Will this search in firstName or lastName. 

OR will this do an `AND` query.

is this currently possible?

or Should i create a computed field to do searches on?

**Comment by kettanaito** [comment, marked as answer]
Hey, @lwhiteley. 

The example above will do an `AND` query, where all conditions must be met by individual data entry in order for it to pass. 

The `OR` queries are currently not supported. You can track its support status and contribute to their implementation/discussion in this issue: https://github.com/mswjs/data/issues/89.
